ctru-rs/src/services/am.rs

```diff
@@ -56,6 +68,55 @@ impl<'a> Title<'a> {
             Ok(info.assume_init())
         }
     }
+
+    pub fn get_smdh(&self) -> crate::Result<Smdh> {
+        // i have no idea how to make this look better
+        let archive_path_data: [u32; 4] =
+            [self.low_u32(), self.high_u32(), self.mediatype as u32, 0x0];
```

Comment:
Do any of the helpers in `src/services/fs.rs` work here? At minimum it looks like there are some helpers already for `Archive`, 

If not, I'd say maybe we should add on there to include SMDH / `"icon"` support , although it might take a bit of finagling to get everything working.